src/rgw/rgw_common.h

```diff
@@ -1527,6 +1527,18 @@ struct rgw_obj_key {
     snprintf(buf, sizeof(buf), "%s[%s]", name.c_str(), instance.c_str());
     return buf;
   }
+
+  // Helper function to find the key of an object.
+  // This function is for handling the null version if specified in the request.
+  // in case of unversioned bucket object key is returned as is.
+  // in case of versioned bucket object key is key[version_id].
+  std::string get_key_name_string() const{
+    if (have_null_instance()) {
+      return name;
+    }
+
+    return to_str();
+  }
```

Comment:
This should be reviewed at ceph repo by rgw devs. Please, create a PR against ceph repo and suggest this change there.

Reply:
@SeagateChaDeepak : We should clear the instance when the user specifies the null version id in Del obj API. This is already done in Rados code, see in: RGWRados::Object::Delete::delete_obj().
 _if (instance == "null") {
    obj.key.instance.clear();
  }_

Similarly, in our Motr sal delete_obj code, we should reset instance when version id is "null".
This won't require any change in common code.

Reply:
@SeagateChaDeepak : Your previous code change (get_object_key_str()) in Motr sal layer looks more feasible then this common code change.